src/hooks/useSubmitQuery.ts

```diff
@@ -170,13 +216,52 @@ export const useSubmitQuery = () => {
         for (const jsonStr of jsonObjects) {
           try {
             const jsonChunk = JSON.parse(jsonStr)
+            console.log('jsonChunk: ', jsonChunk)
 
-            if (jsonChunk.type === 'text' && jsonChunk.content) {
+            if (jsonChunk.type === 'text') {
               accumulatedResponse += jsonChunk.content
               updateLastMessage({
                 role: 'assistant',
                 content: accumulatedResponse,
               })
+            } else if (jsonChunk.type === 'tool_use') {
+              console.log(`${jsonChunk.type}:`, jsonChunk)
+              if (contextConfirmed === null) {
+                contextConfirmed = await showConfirmationModal(
+                  'The assistant is requesting additional context. Do you want to allow this?',
+                )
+              }
+            } else if (jsonChunk.type === 'tool_use_input') {
+              console.log(`${jsonChunk.type}:`, jsonChunk)
```

Comment:
Should we get rid of this?

Reply:
When I tried to remove it didn't work 🤷🏻‍♂️

Reply:
wut lol

Reply:
I meant the console log @umuthopeyildirim 😅

Reply:
lmao :D